### ISSUE TITLE
Vaccu-Stack

### ISSUE THREAD

**Issue by Midaychi** [d3athrow/vgstation13#5508]
A request for a handheld vacuum sorta machine that you can make at the autolathe, and that mommis/engiborgs come with automatically. 

When you use it on a stackable item on the ground, it:

- plays one of various pitch shifted versions of a hand held vacuum sound.
- takes all instances of that or similar stackable object within a range of 2 that aren't already max stacked, and stacks them inside the machine and then drops it at your feet.  After completing a stack, if there's still remaining source material, it'll move on to making a new stack. Etc. (Example: standing in a field of metal sheets, or cut cable pieces. Use on one, singularities it all in and stacks it.)

**Comment by Kurfursten** [comment]
Which ogg?

**Comment by Midaychi** [comment]
There isn't one in the code now, but I found this creative commons 0 one at https://www.freesound.org/people/Huminaatio/sounds/159348/ and edited it into a shorter mono ogg.

https://dl.dropbox.com/s/425v4xcgokg3lzu/vacc.ogg
I can edit it further if its too long or something.

**Comment by Kurfursten** [comment]
Give me :00 to :02

**Comment by Midaychi** [comment]
https://dl.dropbox.com/s/etvwfrptwsxcwg3/vacc_click.ogg  The click?

**Comment by Kurfursten** [comment]
No, all the way to 2 seconds, the new clip is less than 1

**Comment by Midaychi** [comment]
I thought you meant deciseconds. 
Here's just a straight cut:
https://dl.dropbox.com/s/hc619icrq1d54en/vacc_cut.ogg

And here's me attempting to abuse tempo and cleaning up some bits to have a falloff and no click
https://dl.dropbox.com/s/3k931k44jsoufm8/vacc_cut_a.ogg

**Comment by Kurfursten** [comment]
No, I think that second one is really solid

**Comment by Kurfursten** [comment]
I want to make this toolbelt compatible (belt.dm), put it in tools.dm, and use items.dm for the icon. I can't do any of that right now because I have a new PR out on those files, but I'll write this up after that one clears.

Here's a placeholder icon
![icon](https://cloud.githubusercontent.com/assets/9782036/8888998/7667367e-328b-11e5-84cf-607663be5150.png)

**Comment by Midaychi** [comment]
I tried my hand at abusing the placeholder icon to make something. (I'm not super great at this)
https://dl.dropbox.com/s/4smiqjt7jwj2jei/vacc.dmi
No idea if I did the inhand sprites correctly. Honk.
You can pick and choose what you want to use. I just did a whole bunch of different cases. Some coloured ones, some greyscale ones, some all in one and some meant for overlaying, etc.

**Comment by Midaychi** [comment]
Edit whoops, some small pixel issues.
FIxed in this.
https://dl.dropbox.com/s/4smiqjt7jwj2jei/vacc.dmi

**Comment by Intigracy** [comment]
![](http://puu.sh/jdzZb/5e769622c4.png)

**Comment by Kurfursten** [comment]
I'll start work on the code for this after #5515 clears

**Comment by Kurfursten** [comment]
I can't work on this, I'm going on a hiatus from coding. Sorry, @Midaychi, maybe someone else will fill it. It's a nice sprite, and a good sound effect.

**Comment by Kurfursten** [comment]
I'm picking this back up

**Comment by 9600bauds** [comment]
My first attempt at coding this was rather rudimentary
http://a.pomf.cat/bqbwgg.webm

**Comment by Intigracy** [comment]
Can we close this since bauds' alternative fix was put in?

**Comment by clusterfack** [comment]
Yeah I guess
